blog/2024-04-08-reductstore-vs-mongodb/index.md

```diff
@@ -0,0 +1,254 @@
+
+| Chunk Size | Operation | MongoDB, blob/s | ReductStore, blob/s | ReductStore, %    |
+|------------|-----------|-----------------|---------------------|-------------------|
+| 10 KB      | Write     | 258             | 223                 | -14%              |
+|            | Read      | 187             | 195                 | +4%               |
+| 100 KB     | Write     | 145             | 197                 | +36%              |
+|            | Read      | 108             | 145                 | +34%              |
+| 1 MB       | Write     | 32              | 52                  | +63%              |
+|            | Read      | 20              | 29                  | +45%              |
+| 10 MB      | Write     | 4               | 7                   | +75%              |
+|            | Read      | 2               | 4                   | +100%             |
+
+The observed performance trends underscore that while both systems are capable of managing binary data effectively, their throughput varies depending on blob sizes—with ReductStore gaining an edge as size increases.
+
+## Additional Criteria
+
+When selecting a database for blob storage, other criteria than performance must be taken into account:
+
+- **Retention policy**: both ReductStore and MongoDB offer distinct retention strategies. ReductStore uses disk utilization-based policies, essential for edge computing and IoT. MongoDB's time-series collections, on the other hand, can automatically purge documents after a set period of time, useful for managing data lifecycle when dealing with non-blob data such as numerical values or logs in time series scenarios.
+
+- **Replication features**: Replication in ReductStore is append-only, with label-based filtering options. This feature can contribute to an effective data reduction strategy by selectively replicating relevant blobs. MongoDB, on the other hand, offers replication sets for high availability and redundancy, ensuring data integrity and fault tolerance in distributed environments.
+
+- **Cost considerations**: ReductStore's pricing model is based on storage which aligns well with retention policies based on data volume for a predictable cost structure. MongoDB's pricing is based on instance size and usage, which can be more complex to estimate.
+
+## Conclusion
+In conclusion, ReductStore and MongoDB each offer distinct methods for managing and retrieving unstructured time series blob data. MongoDB requires explicit setup with GridFS, while ReductStore simplifies binary data storage into buckets. Performance analysis reveals that while MongoDB is competitive with smaller blob sizes, ReductStore provides superior throughput for larger payloads, making it a compelling choice for heavy-duty time series data scenarios.
```

Comment:
must be updated with the latest results

Reply:
I changed the conclusion accordingly. Let me know if it's ok